src/Nri/Ui/Table/V8.elm

```diff
@@ -0,0 +1,368 @@
+        [ span [ css loadingContentStyles ] [] ]
+
+
+stylesLoadingColumn : Int -> Int -> Style -> List Style
+stylesLoadingColumn rowIndex colIndex width =
+    [ width
+    , property "animation-delay" (String.fromFloat (toFloat (rowIndex + colIndex) * 0.1) ++ "s")
+    ]
+
+
+
+-- HELP
+
+
+tableWithoutHeader : List Style -> List (Column data msg) -> (a -> Html msg) -> List a -> Html msg
+tableWithoutHeader styles columns toRow data =
+    table styles
+        columns
+        [ thead [] [ tr Style.invisible (List.map tableColHeader columns) ]
+        , tableBody toRow data
+        ]
+
+
+tableWithHeader : List Style -> List (Column data msg) -> (a -> Html msg) -> List a -> Html msg
+tableWithHeader styles columns toRow data =
+    table styles
+        columns
+        [ tableHeader columns
+        , tableBody toRow data
+        ]
+
+
+table : List Style -> List (Column data msg) -> List (Html msg) -> Html msg
+table styles columns =
+    Html.table [ css (styles ++ tableStyles columns) ]
+
+
+tableHeader : List (Column data msg) -> Html msg
+tableHeader columns =
+    thead []
+        [ tr [ css headersStyles ]
+            (List.map tableColHeader columns)
+        ]
+
+
+tableColHeader : Column data msg -> Html msg
+tableColHeader (Column header _ width _ sort _ _) =
+    th
+        [ Attributes.scope "col"
+        , css (width :: headerStyles)
+        , Attributes.attribute "aria-sort" <|
+            case sort of
+                Nothing ->
+                    "none"
+
+                Just Ascending ->
+                    "ascending"
+
+                Just Descending ->
+                    "descending"
+        ]
+        [ header ]
+
+
+tableBody : (a -> Html msg) -> List a -> Html msg
+tableBody toRow items =
+    tbody [] (List.map toRow items)
+
+
+
+-- STYLES
+
+
+headersStyles : List Style
+headersStyles =
+    [ -- We use a inset box shadow for a bottom border instead of an actual
+      -- border because with our use of `border-collapse: collapse`, the bottom
+      -- gray border sticks to the table instead of traveling with the header
+      -- when the header has `position: sticky` applied.
+      --
+      -- We add the style on the children instead of on the parent because
+      -- `<tr>` tags do not display box shadows in Safari (although by the time
+      -- you read this, they may have fixed it. Feel free to try again!)
+      Css.Global.children [ Css.Global.th [ boxShadow4 inset (px 0) (px -3) gray75 ] ]
+    , height (px 45)
+    , fontSize (px 15)
+    ]
+
+
+headerStyles : List Style
+headerStyles =
+    [ padding4 (px 11) (px 12) (px 14) (px 12)
+    , textAlign left
+    , fontWeight bold
+    , color gray20
+    ]
+
+
+rowStyles : Bool -> List Style
+rowStyles alternatingRowColors =
+    [ height (px 45)
+    , fontSize (px 14)
+    , color gray20
+    , if alternatingRowColors then
+        pseudoClass "nth-child(odd)"
+            [ backgroundColor gray96 ]
+
+      else
+        Css.batch []
```

Comment:
Note that when there are no alternating row colors we want to show a thin gray border between rows:

Code and screenshot from insights:
https://github.com/NoRedInk/NoRedInk/blob/b54b50b77be0fc6217c4d754f9b3f4fa3f2953a2/monolith/ui/src/Nri/Writing/ClassResults.elm#L840
<img width="351" alt="Screenshot 2024-11-04 at 12 33 23 PM" src="https://github.com/user-attachments/assets/755d4928-4456-4753-8b8c-2fceec8a9246">

Reply:
Noted 👍